### ISSUE TITLE
Kick angles are not interpolated correctly with latest prediction code

### ISSUE THREAD

**Issue by jdolan** [jdolan/quetoo#303]
It looks like the prediction changes for the grapple hook may have broken view kick angle lerp. Set `net_loop_latency 40` and `net_loop_jitter 14` (very reasonable Internet conditions), and watch the view kick when firing a weapon. It jumps from value to value, almost like it's not interpolating at all. Pretty sure this is what everyone will see on public servers, so this is fairly high priority.

**Comment by Paril** [comment]
How this relate to hook? I didn't touch view angle code during  this, however I did notice this bug while developing the hook predict code

**Comment by jdolan** [comment]
Ah, you're right. It was just the first time I had seen this behavior, and when I looked at the prediction code, I saw you had been in there, and hastily assigned blame ;)

I'll setup a server on quetoo.org and test this, see if it's really an issue or what.

**Comment by jdolan** [comment]
So yes, this absolutely happens under normal network conditions. Fux0r.

**Comment by jdolan** [comment]
This has been moved client side now. The server still originates kicks, but the client performs all interpolation and decay.

In working on this, I got to thinking that weapon kicks could be handled completely client side, which might allow them to be smoother (no view bouncing for rapid fire weapons). The cgame could use the attack animation fraction or button state, and apply some static ramp. The ramp amplitude could be specified by the current weapon in the player state stats field (`STAT_WEAPON`).

This way, the server would never have to send weapon kicks as well. The server would only send damage / falling kicks. Would save some more bandwidth as a nice side effect.
